src/NuGet.Packaging/PackageExtraction/PackagePathHelper.cs

```diff
@@ -170,6 +170,25 @@ public static string GetInstalledPackageFilePath(PackageIdentity packageIdentity
 
             return null;
         }
+
+        public static string GetInstalledPackageFolder(PackageIdentity packageIdentity, PackagePathResolver packagePathResolver)
+        {
+            var packageLookupPaths = GetPackageLookupPaths(packageIdentity, packagePathResolver);
+
+            // TODO: Not handling nuspec-only scenarios
+            foreach (var packageLookupPath in packageLookupPaths)
+            {
+                if (Directory.Exists(packageLookupPath) ||
+                    (packageLookupPath.EndsWith(PackagingCoreConstants.NupkgExtension, StringComparison.OrdinalIgnoreCase) &&
+                    File.Exists(packageLookupPath)))
```

Comment:
If packageLookupPath can be a filename, shouldn't you have to return Path.GetDirectoryName(packageLookupPath)? I will simply split the if into 2 and in the case of a file, return Path.GetDirectoryName